### PR TITLE
@craigspaeth Brand partnerships

### PR DESCRIPTION
**Pull request by kanaabe**
Closes https://github.com/artsy/positron/issues/490

brandpartner

### CODE REVIEW

**Comment by craigspaeth** [comment]
Amazing! That was so fast! Impressive! :sparkles: :boom: :fireworks: :sparkles:

File: client/components/layout/templates/sidebar.jade [hunk 1]
```diff
@@ -25,11 +25,11 @@
...
         br
         | Verticals
-      a( href='/organizations'
-         class=(sd.URL.match('/organizations') ? 'is-active' : '') )
+      a( href='/brand-partners'
+         class=(sd.URL.match('/brand-partners') ? 'is-active' : '') )
         include ../public/icons/layout_organizations.svg
         br
-        | Orgs
+        | Brands
```
Comment (kanaabe) [inline client/components/layout/templates/sidebar.jade]: Might not be the best icon to use -- let me know if you have a better one! @owendodd

File: api/apps/brand_partners/model.coffee [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+{ ObjectId } = require 'mongojs'
+{ ARTSY_URL, API_MAX, API_PAGE_SIZE } = process.env
+
+#
+# Schemas
+#
+schema = (->
+  id: @objectId()
+  partner_id: @objectId()
+  slug: @string().allow('', null)
```
Comment (craigspaeth) [inline api/apps/brand_partners/model.coffee]: I don't think this will be necessary b/c the slug is determined by the partner model.

Comment (kanaabe) [inline api/apps/brand_partners/model.coffee]: Good point, so make it searchable [here?](https://github.com/kanaabe/positron/blob/brandpartnership/api/apps/brand_partners/model.coffee#L41)

File: api/apps/brand_partners/index.coffee [hunk 3]
```diff
@@ -0,0 +1,11 @@
+express = require 'express'
+routes = require './routes'
+{ setUser, authenticated, adminOnly } = require '../users/routes'
+
+app = module.exports = express()
+
+app.get '/brand-partners', routes.index
```
Comment (craigspaeth) [inline api/apps/brand_partners/index.coffee]: I think this should be underscore case to keep consistency with our main API & json fields.

Comment (kanaabe) [inline api/apps/brand_partners/index.coffee]: Ah yes, I had it underscored previously, should have gone with my gut. Thanks!

File: client/components/admin_form/index.styl [hunk 4]
```diff
@@ -39,6 +39,13 @@ gutter-size = 30px
 .admin-form-right-col + .admin-form-right-col
   padding-left gutter-size
 
+.admin-form-input-section
+  display inline-block
+  width 50%
+
+.admin-form-input-section + .admin-form-input-section
```
Comment (craigspaeth) [inline client/components/admin_form/index.styl]: :+1:

File: api/apps/brand_partners/index.coffee (hunk 3 above)
Comment (kanaabe) [inline api/apps/brand_partners/index.coffee]: Should we keep the client url as `brand-partners`?

File: test/helpers/fixtures.coffee [hunk 5]
```diff
@@ -78,6 +78,17 @@ module.exports = ->
     'name': 'Artsy Editorial'
     'slug': 'editorial'
     'author_ids': null
+  fixtures.brandPartners =
+    id: '559ff9706b69f6a086a65633'
+    slug: 'VersaceVersace'
```
Comment (craigspaeth) [inline test/helpers/fixtures.coffee]: I do like this slug though!

File: api/apps/brand_partners/index.coffee (hunk 3 above)
Comment (craigspaeth) [inline api/apps/brand_partners/index.coffee]: Hmm no strong opinion here, keeping it dashes on the client works for me.

In general dashes are blessed as the more "SEO friendly" urls b/c Google used to not treat underscores as word separators, but that 1. Obviously doesn't matter for a non-website app like Writer and 2. I think Google treats both as separators  these days anyways.

File: api/apps/brand_partners/model.coffee (hunk 2 above)
Comment (craigspaeth) [inline api/apps/brand_partners/model.coffee]: So in that case we can just do `query = { _id: ObjectId(id) }` b/c we don't need to support `/api/brand_partners/:slug` we should instead just query based on the partner id (which you support [here](https://github.com/kanaabe/positron/blob/brandpartnership/api/apps/brand_partners/model.coffee#L34) but I think you need to add a line that converts string -> object id [here](https://github.com/kanaabe/positron/blob/brandpartnership/api/apps/brand_partners/model.coffee#L48)) e.g. `/api/brand_partners?partner_id=:id`.

Comment (craigspaeth) [inline api/apps/brand_partners/model.coffee]: Example of converting string -> objectId in where function: https://github.com/kanaabe/positron/blob/master/api/apps/articles/model.coffee#L138 ...going to submit that patch to Joi-objectid some day!

**Comment by craigspaeth** [comment]
Looks good! Mainly concerned about the comments on snake_case and ensuring querying `/api/brand_partners?partner_id=` works. Then it's good to merge!

**Comment by craigspaeth** [comment]
Well done!

**Pull request merged**